ocpp/v16/enums.py

```diff
@@ -619,3 +631,156 @@ class ValueFormat(str, Enum):
+    manufacturer_root_certificate = "ManufacturerRootCertificate"
+
+
+class DeleteCertificateStatus(str, Enum):
+    """
+    DeleteCertificateStatusEnumType is used by: DeleteCertificate.conf
+    """
+    accepted = "Accepted"
+    failed = "Failed"
+    not_found = "NotFound"
+
+
+class SignedFirmwareStatus(str, Enum):
+    """
+    FirmwareStatusEnumType is used by: SignedFirmwareStatusNotification.req
+    """
+    certificate_verified = "CertificateVerified"
+    downloaded = "Downloaded"
+    download_failed = "DownloadFailed"
+    downloading = "Downloading"
+    download_scheduled = "DownloadScheduled"
+    download_paused = "DownloadPaused"
+    idle = "Idle"
+    installation_failed = "InstallationFailed"
+    installing = "Installing"
+    installed = "Installed"
+    install_rebooting = "InstallRebooting"
+    install_scheduled = "InstallScheduled"
+    install_verification_failed = "InstallVerificationFailed"
+    invalid_signature = "InvalidSignature"
+    invalid_certificate = "InvalidCertificate"
+    revoked_certificate = "RevokedCertificate"
+    signature_verified = "SignatureVerified"
+
+
+class GenericStatus(str, Enum):
+    """
+    Generic message response status
+    """
+    accepted = "Accepted"
+    rejected = "Rejected"
+
+
+class GetInstalledCertificateStatus(str, Enum):
+    """
+    GetInstalledCertificateStatusEnumType is used by:
+    GetInstalledCertificateIds.conf
+    """
+    accepted = "Accepted"
+    not_found = "NotFound"
+
+
+class HashAlgorithm(str, Enum):
+    """
+    HashAlgorithmEnumType is used by: CertificateHashDataType
+    """
+    sha256 = "SHA256"
+    sha384 = "SHA384"
+    sha512 = "SHA512"
+
+
+class Log(str, Enum):
+    """
+    LogEnumType is used by GetLog.req
+    """
+    diagnostics_log = "DiagnosticsLog"
+    security_log = "SecurityLog"
+
+
+class LogStatus(str, Enum):
+    """
+    LogStatusEnumType is used by: GetLog.conf
+    """
+
+    accepted = "Accepted"
+    rejected = "Rejected"
+    accepted_canceled = "AcceptedCanceled"
+
+
+class ExtendedMessageTrigger(str, Enum):
+    """
+    MessageTriggerEnumType is used by: ExtendedTriggerMessage.req
+    """
+
+    boot_notification = "BootNotification"
+    log_status_notification = "LogStatusNotification"
+    firmware_status_notification = "FirmwareStatusNotification"
+    heartbeat = "Heartbeat"
+    meter_values = "MeterValues"
+    sign_charge_point_certificate = "SignChargePointCertificate"
+    status_notification = "StatusNotification"
+
+
+class ExtendedTriggerMessageStatus(str, Enum):
+    """
+    TriggerMessageStatusEnumType is used by: ExtendedTriggerMessage.conf
+    """
+
+    accepted = "Accepted"
+    rejected = "Rejected"
+    not_implemented = "NotImplemented"
+
+
+class UpdateFirmwareStatus(str, Enum):
+    """
+    UpdateFirmwareStatusEnumType is used by: SignedUpdateFirmware.conf
+    """
+
+    accepted = "Accepted"
+    rejected = "Rejected"
+    accepted_canceled = "AcceptedCanceled"
+
```

Comment:
These are missing:
* InvalidCertificate
* RevokedCertificate

Reply:
Well spotted. These seem to be missing from edition 2 FINAL, 2017-09-28 specification but do exists in SignedUpdateFirmwareResponse.json schema.

Reply:
I am not sure that was labeled as final version, but the one I got, was the one officially released
[OCPP 1.6 security whitepaper edition 2.pdf](https://github.com/mobilityhouse/ocpp/files/6435221/OCPP.1.6.security.whitepaper.edition.2.pdf)